import java.util.Scanner;
  //   -->  VIDEO_80   <--
 // EXCEPTION THROW CODE 

//public class chw_14_vid_78 {
//    public static void main(String[] args) {
//        Scanner sc= new Scanner(System.in);
//         int a= sc.nextInt();
//        // Throw the Exceptions Error
//        System.out.println("Interger parts Divided of 1000 by k is: "+1000/a);
//    }
//}


     // EXCEPTION IN JAVA -- FIX EXCEPTION

public class chw_14_vid_78 {
    public static void main(String[] args) {
        Scanner sc= new Scanner(System.in);
         int a= 1000;
         int b=sc.nextInt();
         try{
             int c= a/b;
             System.out.println("Interger parts Divided of 1000 by k is: "+c);
         }
         catch (Exception e){
             System.out.println(e);
             System.out.println("You are Fail The Devided Reason");
         }
        System.out.println("The End of Program");
    }
}


   ----->     #VIDEO NUMBER 82  <-----
     // HANDLINH SPECIFIC EXCEPTIONS

     class SpecificExceptions extends Thread{
         public void run(){
             int[]marks=new int[3];
              marks[0]=7;
              marks[1]=56;
              marks[2]=6;
              Scanner sc= new Scanner(System.in);
             System.out.println("Enter The Arry Index");
                  int n=sc.nextInt();
                  int m= sc.nextInt();
                  try {
                          System.out.println("The Value of number is :" + marks[n] / m);
                      }
                  catch(ArithmeticException e){
                          System.out.println(".ArithmeticException Exception :");
                          System.out.println(e); // jo kami hota hai ushe show karta hai / by zero
                  }
                  catch(ArrayIndexOutOfBoundsException e){
                          System.out.println("ArrayIndexOutOfBoundsException occured!");
                          System.out.println(e);
                      }
                 catch(Exception e){
                          System.out.println("Some other exception occured! ");
                          System.out.println(e);
                  }
         }
     }
public class chw_14_vid_78 {
    public static void main(String[] args) {
        SpecificExceptions obj= new SpecificExceptions();
        obj.start();

    }
}

-----------------------------------------------------------------------------
 ----->  // NESTED TRY- CATCH IN JAVA <-----

     class nestedtry extends Thread{
         public void run() {
             int[] marks = new int[3];
             marks[0] = 7;
             marks[1] = 56;
             marks[2] = 6;

             Scanner sc = new Scanner(System.in);
             boolean flag=true;
             while(flag){
             System.out.println("Enter The Arry Index");
             int ind = sc.nextInt();

             try {
                 System.out.println("Nested Try 1 vid 82");
                 try {
                     System.out.println("Marks of Number is  : "+marks[ind]);
                     flag=false;

                 } catch(ArrayIndexOutOfBoundsException e){
                     System.out.println("Sorry this index does not exist");
                     System.out.println("Exception in leavel 1");
                 }
             }
            catch (Exception e) {
                System.out.println("Exception in leavel 2");
                // System.out.println(e);
            }
             }
             System.out.println("Thanks for Using Program");
         }
     }

public class chw_14_vid_78 {
    public static
    void main(String[] args) {
        nestedtry obj = new nestedtry();
        obj.start();
    }
}
----------------------------------------------------------------------------------
  //---> EXCEPTION CLASS  <----
#VIDEO NO 83 

import java.util.*;
class MyException extends Exception{
    public void run(){
        System.out.println(" I am run method");
    }
    public String toString(){
        return " I am toString class";
    }
    public String getMessage(){
        return " I am getmessage Class";
    }
}

public class chapter_14 {
    public static void main(String[] args) {
        int age;
        Scanner sc = new Scanner(System.in);
        boolean flag=true;
        while(flag) {
            age = sc.nextInt();
            if (age < 15) {
                try {
//           System.out.println("Sorry you can not Drive ");
                    throw new ArithmeticException("Sorry you can not Drive  ");

                } catch (Exception e) {
                    System.out.println(e.getMessage());
                   // System.out.println(e.toString());
                    System.out.println(e); // Only print(e) Throw toString ;
                     e.printStackTrace();
                }
                System.out.println("finish");
             ;
            } else {
                System.out.println("you can drive");
                flag=false;
            }
        }

        //--->Upper wala code me Exception nhi aayega tabi niche wale code run hoga
        System.out.println("Enter Number b");
         int b=sc.nextInt();
        try{
            System.out.println(100/b);
            System.out.printf(" corect input");
        }
        catch(Exception e){
            System.out.println("Throw Exception .ArithmeticException: / by zero ");
            System.out.println(e);
        }
    }
}

--------------------------------------------------------------------------------------------
  //  ---  THE THROWS EXCEPTION  ---- #VID_84
import java.util.Scanner;
//--->CODE FOR DRIVING CAR ARE NOT DRIVE <---// made by me
// Enter Age and you find the Eligiblility Chartery
// Only wrong 4 time Because set Charteriya (i)
// Agar ap age<18 Dalte hai to custom Exception Throw karega
/*
class cheakDrivecar extends Exception{
    public String toString(){
        return " You able to Drive";
    }
    public String getMessage(){
        return "You can not Drive";
    }
}
public class Custom_Exception {
    public static int yourage(int age) throws cheakDrivecar{
        if(age<18 ||  age>140){
            throw new cheakDrivecar();
        }
        System.out.println("Yes you can Drive ");
        return age ;

    }
    public static void main(String[] args) {   //throws cheakDrivecar {
        Scanner sc= new Scanner(System.in);
        boolean flag=true;
        int i=0;
         while(flag && i<5) { //i<5 ka matlab ap 5 bar hi wrong Enter kar sakte hai
             System.out.println("Enter your age");
             int num = sc.nextInt();
             try {
                 int a = yourage(num);
                 System.out.println(a);
                flag=false;
             } catch (Exception e) {
                 e.getStackTrace();
                 System.out.println("Sorry Your age is age>18 So you can not Drive");
                 System.out.println(e.getStackTrace());
                 System.out.println(e.getMessage());
               i++;
             }

             System.out.println("Finish");

         }
    }
      }
// Agar apko Aise print karane ho to ap main ke pass -> throws cheakDrivecar { <- use karenge
        //int a=yourage(13);
         //   System.out.println(a);
        //
*/

----------------------------------------------------------------------
 //---> JAVA FINALLY BLOCK <----
     //#Vid_85
public class Custom_Exception {
    public static int greet(){
    Scanner sc=new Scanner(System.in);
        System.out.println("Enter First num: ");
    int a= sc.nextInt();
        System.out.println("Enter First Second: ");
    int b= sc.nextInt();
    try{
        int result=a/b;
        return result;
    }
    catch( Exception e) {
        System.out.println(e);
    }
    return -1; // Your choice put any number
    }
    public static void main(String[] args) {
        int ab=greet();
//        System.out.println(ab);

            try{
                System.out.println(ab);
            }
            catch(Exception e){
                System.out.println(e);
            }
            finally{
                System.out.println(" I am finally Block..This is the end of This funcation");

        }

    }

}



